packages/client/wallets/smart-wallet/src/blockchain/wallets/SendTransactionService.ts

```diff
@@ -0,0 +1,191 @@
+import type { SmartAccountClient } from "permissionless";
+import type { EntryPoint } from "permissionless/_types/types";
+import type { SmartAccount } from "permissionless/accounts";
+import {
+    type Abi,
+    type Address,
+    BaseError,
+    type Chain,
+    ContractFunctionRevertedError,
+    type Hex,
+    type PublicClient,
+    type TransactionReceipt,
+    type Transport,
+} from "viem";
+
+import { CrossmintSDKError, WalletErrorCode } from "@crossmint/client-sdk-base";
+
+export type TransactionServiceTransactionRequest = {
+    address: Address;
+    abi: Abi;
+    functionName: string;
+    args: any;
+    value?: bigint;
+};
+
+export type SendTransactionFailureStage = "simulation" | "send" | "confirmation" | "execution";
+
+/**
+ * Error thrown when a transaction fails to send.
+ * @param viemError The error thrown by the viem client. See https://viem.sh/docs/glossary/errors.html
+ */
+export class EVMSendTransactionError extends CrossmintSDKError {
+    constructor(
+        message: string,
+        public readonly viemError: BaseError,
+        public readonly stage: SendTransactionFailureStage,
+        code = WalletErrorCode.SEND_TRANSACTION_FAILED
+    ) {
+        super(message, code);
+    }
+}
+
+/**
+ * Error thrown when a transaction fails due to a contract execution error.
+ * @param viemError The error thrown by the viem client. See https://viem.sh/docs/glossary/errors.html
+ * @param revertError The revert error from viem containing the reason for the revert.
+ * @param revertError.reason The reason for the revert.
+ * @param revertError.data The decoded revert error data.
+ * @example
+ * try {
+ *   await wallet.sendTransaction({
+ *     address: contractAddress,
+ *     abi,
+ *     functionName: "mintNFT",
+ *     args: [recipientAddress],
+ *   });
+ * } catch (e) {
+ *   if (e instanceof SendTransactionExecutionRevertedError) {
+ *     alert(`Transaction reverted: ${e.revertError.reason}`);
+ *   }
+ *   throw e;
+ * }
+ */
+export class EVMSendTransactionExecutionRevertedError extends EVMSendTransactionError {
```

Comment:
Any downside to having something more concise, maybe `EVMTransactionRevertedError` instead?

Reply:
Base error could be `EVMTransactionError`

Reply:
My personal opinion is that clarity is more important than brevity, worth mentioning that viem seems to follow them same philosophy with their errors: https://viem.sh/docs/glossary/errors.html

Reply:
I'm not sure the `send` or `execution` add more clarity here, but I could just be too used to hearing all the terminology often.